source/common/http/codec_client.h

```diff
@@ -173,8 +173,12 @@ class CodecClient : Logger::Loggable<Logger::Id::client>,
     CodecReadFilter(CodecClient& parent) : parent_(parent) {}
 
     // Network::ReadFilter
-    Network::FilterStatus onData(Buffer::Instance& data, bool) override {
+    Network::FilterStatus onData(Buffer::Instance& data, bool end_stream) override {
       parent_.onData(data);
+      // Reset the codec client on end stream when HTTP1
+      if (end_stream && parent_.protocol() <= Protocol::Http11) {
```

Comment:
One request, can we comment here that this will result in the connecting looking like it closed locally when it is functionally closed remotely?

I'd be mildly inclined to either put in a check to see if the connection has half close enabled here, or to add a test that in the normal case when the connection is closed we get the close event before an onData(... true) just to make sure the mis-attribution only applies for our CONNECT path.

Reply:
Ok, I'll start by adding the comment.

Regarding the check, I took a quick look but at first glance, I did not see an easy way of adding it. I will take another look.

Reply:
Well, adding a method `isHalfCloseEnabled` on the CodecClient would make it.